lua/ide/components/bufferlist/component.lua

```diff
@@ -0,0 +1,247 @@
+local base = require("ide.panels.component")
+local libwin = require("ide.lib.win")
+local logger = require("ide.logger.logger")
+local icon_set = require("ide.icons").global_icon_set
+local commands = require("ide.components.bufferlist.commands")
+
+local BufferListComponent = {}
+
+local config_prototype = {
+	disabled_keymaps = false,
+	keymaps = {
+		edit = "<CR>",
+		edit_split = "s",
+		edit_vsplit = "v",
+		close = "d",
+	},
+}
+
+local function get_listed_bufs()
+	return vim.tbl_filter(function(bufnr)
+		return vim.bo[bufnr].buflisted and vim.api.nvim_buf_is_loaded(bufnr) and vim.api.nvim_buf_get_name(bufnr) ~= ""
+	end, vim.api.nvim_list_bufs())
+end
+
+-- Get the names of all current listed buffers
+local function get_current_filenames()
+	return vim.tbl_map(vim.api.nvim_buf_get_name, get_listed_bufs())
+end
+
+-- Get unique name for the current buffer
+local function get_unique_filename(filename)
+	local filenames = vim.tbl_filter(function(filename_other)
```

Comment:
Ditto: Should this be in libbuf? 

Also why exactly is this necessary? You can use the buffer IDs + Name to obtain uniqueness?

Reply:
This is the way most other buffer list plugins handle this.

Reply:
cool :) - wasn't aware.